contracts/gateway/src/verifier.rs

```diff
@@ -0,0 +1,109 @@
+use crate::contract::Error;
+use axelar_wasm_std::VerificationStatus;
+use connection_router::state::CrossChainId;
+use connection_router::Message;
+use cosmwasm_std::{to_binary, Addr, QuerierWrapper, QueryRequest, WasmMsg, WasmQuery};
+use error_stack::{Result, ResultExt};
+use serde::de::DeserializeOwned;
+use std::collections::HashMap;
+
+pub struct Verifier<'a> {
+    pub querier: QuerierWrapper<'a>,
+    pub address: Addr,
+}
+
+impl Verifier<'_> {
+    fn query<U: DeserializeOwned + 'static>(
+        &self,
+        msg: &aggregate_verifier::msg::QueryMsg,
+    ) -> Result<U, Error> {
+        self.querier
+            .query(&QueryRequest::Wasm(WasmQuery::Smart {
+                contract_addr: self.address.to_string(),
+                msg: to_binary(&msg).expect("msg should always be serializable"),
+            }))
+            .change_context(Error::QueryVerifier)
+    }
+
+    fn execute(&self, msg: &aggregate_verifier::msg::ExecuteMsg) -> WasmMsg {
+        WasmMsg::Execute {
+            contract_addr: self.address.to_string(),
+            msg: to_binary(msg).expect("msg should always be serializable"),
+            funds: vec![],
+        }
+    }
+
+    pub(crate) fn messages_status(
+        &self,
+        msgs: Vec<Message>,
+    ) -> Result<impl Iterator<Item = (Message, VerificationStatus)>, Error> {
+        let response: Vec<(CrossChainId, VerificationStatus)> =
+            self.query(&aggregate_verifier::msg::QueryMsg::GetMessagesStatus {
+                messages: msgs.clone(),
+            })?;
+
+        let status_by_id = response.into_iter().collect::<HashMap<_, _>>();
+
+        Ok(msgs.into_iter().map(move |msg| {
+            let status = status_by_id
+                .get(&msg.cc_id)
+                .copied()
+                .unwrap_or(VerificationStatus::None);
```

Comment:
What's the difference between `VerificationStatus::NotFound` and `VerificationStatus::None`?

Reply:
`None` means it's never been verified, `NotFound` means that it wasn't found on the source chain. We need to rename that in a separate PR